### DISCUSSION TITLE
DGS CodeGen all fields

### DISCUSSION THREAD

**Discussion by andresmonc** [Netflix/dgs-framework#1519] [Q&A]
Is there a way to set all fields on a projection root? I have a huge graphql query that returns an object with a large amount of fields and many nested objects. Is there a way to just return everything?

 new DealProjectionRoot<>().id()...;

**Comment by srinivasankavitha** [comment]
Unfortunately, we don't have that feature implemented yet. While it would be nice, in graphql, you do have to select all the fields explicitly in the request as well and the API mirrors that functionality. Adding an allFields API to the projection root is an option, but it does introduce conflicts in case there is a schema that uses the same name for their field, so we will need to be careful about that.
